satpy/readers/hdfeos_mod35.py

```diff
@@ -0,0 +1,434 @@
+
+        array.attrs = info
+
+        return array
+
+#right_shift = da.array.frompyfunc(np.right_shift)
+#left_shift = da.array.frompyfunc(np.left_shift)
+
+
+def bits_stripping(bit_start, bit_count, value):
+    """Extract specified bit from bit representation of integer value.
+
+    Parameters
+    ----------
+    bit_start : int
+        Starting index of the bits to extract (first bit has index 0)
+    bit_count : int
+        Number of bits starting from bit_start to extract
+    value : int
+        Number from which to extract the bits
+
+    Returns
+    -------
+    int
+        Value of the extracted bits
+
+    """
+
+    bitmask = pow(2, bit_start + bit_count) - 1
+
+    return np.right_shift(da.bitwise_and(value, bitmask), bit_start)
+
+
+def BitShiftCombine(byte1, byte2):
+    """
+    Combine two 8 bit integers to one 16 bit integer by
+    concatenating
+
+    Parameters
+    ----------
+    byte1, byte2 : np.int
+        8 bit integers two combine
+
+    Returns
+    -------
+    np.uint16
+        16 bit integer with byte1 as the first 8 bit and byte2 as the second 8 bit
+    """
+    # make sure byte1 is 16 bit
+    twoByte = np.uint16(byte1)
+    # shift bits to the left
+    twoByte = np.left_shift(twoByte, 8)  # twoByte << 8
+    # concatenate the two bytes
+    twoByte = da.bitwise_or(twoByte, byte2).astype(np.uint16)  # casting = "no")
+
+    return twoByte
+
+
+def bit_strip_250m_mask(inputArray):
+    """
+    Create 250m cloud mask from the last 16 bits of the modis 48 bit array.
+
+    Takes each array element decodes the bits and distributes
+    them to new array with 4 times the resolution in x and y direction
+    of the input array (each input array element is replaced by 16 elements
+    so to speak).
+
+    Parameters
+    ----------
+    inputArray : np.uint16
+        input array which to transform
+
+
+    Returns
+    -------
+    np array
+        Array with increased resolution and binary mask based on bits of input array
+
+    """
+
+    # create 4x4 array with numbers from 1 to 16 (count of bits)
+    pix = da.arange(0, 16, chunks = (1, )).reshape((4, 4)).astype(np.uint16)
+    # tile the 4x4 array the size of the input array dimensions
+    pix = da.tile(pix, (inputArray.shape[0], inputArray.shape[1]))
+    # "interpolate" input array with nearest neighbour with resolution increase 4
+    data = da.repeat(np.repeat(inputArray, 4, axis=0), 4, axis=1)
+    # decode the corresponding bits to get the 250m mask
+    mask250 = bits_stripping(pix, 1, data)
+
+    return mask250
+
+
+if __name__ == '__main__':
+    from satpy.utils import debug_on
+    debug_on()
+    br = HDFEOSBandReader(
+        '/media/droplet_data/data/modis_lcrs_2005_2006/mod35/2005/MOD35_L2.A2005001.0740.006.2014340132609.hdf')
+    gr = HDFEOSGeoReader(
+        '/media/droplet_data/data/modis_lcrs_2005_2006/mod03/2005/MOD03.A2005001.0740.006.2012274114305.hdf')
```

Comment:
It's good to test with real data. We have a framework for this here: https://github.com/pytroll/satpy/blob/master/satpy/tests/features/feature-real-load-process-write.feature
Maybe you can use that ?

Reply:
I'll have a look at it, thanks.